cmake/CMakeLists.txt

```diff
@@ -803,6 +803,12 @@ if (onnxruntime_USE_DML)
     list(APPEND ORT_PROVIDER_FLAGS  -DUSE_DML=1)
     list(APPEND ORT_PROVIDER_CMAKE_FLAGS -Donnxruntime_USE_DML=1)
     list(APPEND ONNXRUNTIME_PROVIDER_NAMES dml)
+    if(onnxruntime_DML_ENABLE_SERIALIZATION)
+      list(APPEND ORT_PROVIDER_FLAGS -DDML_ENABLE_SERIALIZATION=1)
+      if(onnxruntime_DML_ENABLE_SERIALIZATION_DEBUG)
```

Comment:
Would it make sense to allow only the `onnxruntime_DML_ENABLE_SERIALIZATION_DEBUG` flag to be provided, independently of the `onnxruntime_DML_ENABLE_SERIALIZATION` flag?

Reply:
Actually we can. As of now I kept it as following:
- `onnxruntime_DML_ENABLE_SERIALIZATION` will be required to serialize the Dml partition
- `onnxruntime_DML_ENABLE_SERIALIZATION_DEBUG` is optional to only trigger the deserializer to verify the serialized graph.

But we can actually make these 2 flag independent of each other to avoid passing 2 build flags and achieve the above functionality.